flink-table/flink-table-planner/src/main/java/org/apache/flink/table/expressions/rules/ExpandColumnFunctionsRule.java

```diff
@@ -193,7 +190,7 @@ public ColumnsExpressionExpander(List<UnresolvedReferenceExpression> inputFieldR
 
 		@Override
 		protected List<UnresolvedReferenceExpression> defaultMethod(Expression expression) {
-			throw new TableException(
+			throw new ValidationException(
 				String.format("The parameters of %s() or %s() only accept column name or column " +
```

Comment:
One additional location that looks like inconsistent regarding exceptions. Is the `Preconditions.checkArgument`. Usually this method is for checking if we (the Flink developers) did everything correct how every this check is also validating user input, right?

Reply:
Maybe `Preconditions.checkArgument` can also be used to validate user input, such as the checkArgument in `org.apache.flink.streaming.api.environment.StreamExecutionEnvironment`.  However, I think it also makes sense to change the `Preconditions.checkArgument` to ValidationException. I will update the PR according to your suggestions.